src/extension.ts

```diff
@@ -1,234 +1,235 @@
-                    telemetryReporter.sendTelemetryEvent("command/attach/devtools", telemetryProps);
-                    const runtimeConfig = getRuntimeConfig(config);
-                    DevToolsPanel.createOrShow(context, telemetryReporter, selection.detail, runtimeConfig);
-                }
-            }
-        } else {
-            telemetryReporter.sendTelemetryEvent("command/attach/error/no_json_array", telemetryProps);
-        }
-    } while (useRetry && Date.now() - startTime < timeout);
-}
-
-export async function launch(context: vscode.ExtensionContext, launchUrl?: string, config?: Partial<IUserConfig>) {
-    if (!telemetryReporter) {
-        telemetryReporter = createTelemetryReporter(context);
-    }
-
-    const telemetryProps = { viaConfig: `${!!config}` };
-    telemetryReporter.sendTelemetryEvent("command/launch", telemetryProps);
-
-    const { hostname, port, defaultUrl, userDataDir } = getRemoteEndpointSettings(config);
-    const url = launchUrl || defaultUrl;
-    const target = await openNewTab(hostname, port, url);
-    if (target && target.webSocketDebuggerUrl) {
-        // Show the devtools
-        telemetryReporter.sendTelemetryEvent("command/launch/devtools", telemetryProps);
-        const runtimeConfig = getRuntimeConfig(config);
-        DevToolsPanel.createOrShow(context, telemetryReporter, target.webSocketDebuggerUrl, runtimeConfig);
-    } else {
-        // Launch a new instance
-        const browserPath = await getBrowserPath(config);
-        if (!browserPath) {
-            telemetryReporter.sendTelemetryEvent("command/launch/error/browser_not_found", telemetryProps);
-            vscode.window.showErrorMessage(
-                "Microsoft Edge could not be found. " +
-                "Ensure you have installed Microsoft Edge " +
-                "and that you have selected 'default' or the appropriate version of Microsoft Edge " +
-                "in the extension settings panel.");
-            return;
-        } else {
-            // Here we grab the last part of the path (using either forward or back slashes to account for mac/win),
-            // Then we search that part for either chrome or edge to best guess identify the browser that is launching.
-            // If it is one of those names we use that, otherwise we default it to "other".
-            // Then we upload just one of those 3 names to telemetry.
-            const exeName = browserPath.split(/\\|\//).pop();
-            const match = exeName!.match(/(chrome|edge)/gi) || [];
-            const knownBrowser = match.length > 0 ? match[0] : "other";
-            const browserProps = { exe: `${knownBrowser.toLowerCase()}` };
-            telemetryReporter.sendTelemetryEvent("command/launch/browser", browserProps);
-        }
-        browserInstance = await launchBrowser(browserPath, port, url, userDataDir);
-        await attach(context, url, config);
-    }
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
not sure why but the file shows all the content as new/changed which makes reviewing it really difficult. Could you double check and scope the review to the changes?

Reply:
It looks like this is happening whenever we use Github's rebase... Not sure why, but I'll try fixing this.

Reply:
Okay fixed and ready for review